source/docs/software/trajectory-end-to-end/creating-following-trajectory.rst

```diff
@@ -0,0 +1,153 @@
+
+.. tabs::
+
+  .. group-tab:: Java
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/ramsetecommand/RobotContainer.java
+      :language: java
+      :lines: 96-103
+      :linenos:
+      :lineno-start: 96
+
+  .. group-tab:: C++ (Source)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/examples/RamseteCommand/cpp/RobotContainer.cpp
+      :language: c++
+      :lines: 47-51
+      :linenos:
+      :lineno-start: 47
+
+Notice that we set the maximum voltage to 10V, rather than the nominal battery voltage of 12V.  This gives us some "headroom" to deal with "voltage sag" during operation.
+
+Creating the Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that we have our voltage constraint, we can create our ``TrajectoryConfig`` instance, which wraps together all of our path constraints:
+
+.. tabs::
+
+  .. group-tab:: Java
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/ramsetecommand/RobotContainer.java
+      :language: java
+      :lines: 105-111
+      :linenos:
+      :lineno-start: 105
+
+  .. group-tab:: C++ (Source)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/examples/RamseteCommand/cpp/RobotContainer.cpp
+      :language: c++
+      :lines: 53-59
+      :linenos:
+      :lineno-start: 53
+
+Generating the Trajectory
+-------------------------
+
+With our trajectory configuration in hand, we are now ready to generate our trajectory.  For this example, we will be generating a "clamped cubic" trajectory - this means we will specify full robot poses at the endpoints, and positions only for interior waypoints (also known as "knot points").  As elsewhere, all distances are in meters.
+
+.. tabs::
+
+  .. group-tab:: Java
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/ramsetecommand/RobotContainer.java
+      :language: java
+      :lines: 113-126
+      :linenos:
+      :lineno-start: 113
+
+  .. group-tab:: C++ (Source)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/examples/RamseteCommand/cpp/RobotContainer.cpp
+      :language: c++
+      :lines: 61-70
+      :linenos:
+      :lineno-start: 61
+
+Creating the RamseteCommand
+---------------------------
+
+.. note:: It is very important that the robot begin the trajectory-following at a pose corresponding to the start pose of the trajectory.  As our example is an autonomous command, the robot will be reliably starting at a position of ``(0,0)`` with a heading of ``0``.  When in doubt, it is always possible to reset the odometry to a set position.
```

Comment:
As mentioned in https://github.com/wpilibsuite/allwpilib/pull/2180, resetting the odometry is probably not the right advice here.

Reply:
Until/unless we add a feature for applying a change-of-coordinates to a full trajectory, I think this has to stay as-is.

Reply:
You need to set the odometry to something at least, because the robot basically never starts at (0, 0) on the field.

Reply:
There's nothing wrong with doing paths that are entirely robot-relative for a simple demo, I think.